app/models/concerns/git_hub_repoable.rb

```diff
@@ -43,7 +45,18 @@ def destroy_github_repository
   #
   def push_starter_code
     return true unless starter_code_repo_id
-    PushStarterCodeJob.perform_later(creator, github_repo_id, starter_code_repo_id)
+
+    client = creator.github_client
+
+    assignment_repository   = GitHubRepository.new(client, github_repo_id)
+    starter_code_repository = GitHubRepository.new(client, starter_code_repo_id)
+
+    begin
+      assignment_repository.get_starter_code_from(starter_code_repository)
+    rescue
+      destroy_github_repository
```

Comment:
@tarebyte what if we get an exception here?

We should probably do something smart like tell the user what happened and ask them to report a bug or give them a way to fix the situation.